src/blockmap.jl

```diff
@@ -0,0 +1,297 @@
+struct BlockMap{T,As<:Tuple{Vararg{LinearMap}},Rs<:Tuple{Vararg{Int}}} <: LinearMap{T}
+    maps::As
+    rows::Rs
+    function BlockMap(maps::R, rows::S) where {T, R<:Tuple{Vararg{LinearMap{T}}}, S<:Tuple{Vararg{Int}}}
+        new{T,R,S}(maps, rows)
+    end
+end
+
+cumsize_h(maps::Tuple{Vararg{LinearMap}}) = cumsum([1, map(m -> size(m, 2), maps)...,])
```

Comment:
I am still confused why this returns `1, rowsize of map1 + 1, rowsize of map 1 and 2 + 1, ...`
rather than `0, rowsize of map1, rowsize of map1 and 2, ...`, i.e. about the `+1` offset. I see the advantage, but find `cumsize` a rather confusing name. How about `firstindices` or `startindices`.

Reply:
Regarding the first comment: `cumsum` of a tuple does not work in Julia Base. I accidently had `TupleTools` loaded while trying this, which seems to commit type piracy and provides a definition of `cumsum` for tuples.

Reply:
I think I tried the tuple version, but failed as you noticed. The offset by 1 is because it's a cumulative sum: the first cumsum entry is simply the first entry, so 1; the second is the sum of `rowsize of map1` plus the first entry, which is 1; etc. I'm fine with `firstindices`. The `cumsize` is probably another legacy of my first attempts to adopt `BlockArrays.jl`.